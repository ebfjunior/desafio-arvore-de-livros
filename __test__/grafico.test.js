import React from 'react';

import Periodos from '../src/business/periodos';
import Grafico from '../src/components/grafico';

describe('Componente Gráfico', () => {
  it('Deve renderizar adequadamente', () => {
    const wrapper = mount(<Grafico/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('State Períodos deve ser inicializado corretamente', () => {
    const wrapper = mount(<Grafico/>);
    expect(Object.keys(wrapper.state().periodos.processados).length).toEqual(175);
  });

  it('State Data Final deve ser inicializado corretamente', () => {
    const wrapper = mount(<Grafico/>);
    expect(wrapper.state().data_final).toEqual("2017-06-08");
  });

  it('State Data Inicial deve ser um ano antes que a Data Final', () => {
    const wrapper = mount(<Grafico/>);
    expect(wrapper.state().data_inicial).toEqual("2016-06-08");
  });
});