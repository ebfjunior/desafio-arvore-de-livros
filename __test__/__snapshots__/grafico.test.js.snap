// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Gráfico Deve renderizar adequadamente 1`] = `
<Grafico>
  <span>
    Hello, Árvore de Livros
  </span>
</Grafico>
`;
